File: store_application/lib/model/address.dart
class Address {
  // int? id;
  double latitude = 0;
  double longitude = 0;
  String country = "";
  String city = "";
  String area = "";
  String street = "";

  // String? buildingNo = "1";
  //
  // Address({
  //   // this.id,
  //    this.area,
  //   // this.buildingNo,
  //    this.city,
  //    this.country,
  //    this.latitude,
  //    this.longitude,
  //    this.street,
  // });
  //
  // factory Address.fromJson(Map<String, dynamic> json) {
  //   return Address(
  //     // id: json["id"] ?? 1,
  //     latitude: json["latitude"] ?? 33.222,
  //     longitude: json["longitude"] ?? 34.223,
  //     country: json["country"] ?? "",
  //     city: json["city"] ?? "",
  //     area: json["area"] ?? "",
  //     street: json["street"] ?? "",
  //     // buildingNo: json["buildingNo"] ?? "",
  //   );
  // }

  Map<String, dynamic> toJson() => {
        "latitude": latitude,
        "longitude": longitude,
        "country": country,
        "city": city,
        "area": area,
        "street": street,
        // "building_no": buildingNo,
      };
}

File: ecommerce-backend/app/Models/Address.ts
import { DateTime } from 'luxon'
import { BaseModel, BelongsTo, belongsTo, column } from '@ioc:Adonis/Lucid/Orm'
import City from './City'

export default class Address extends BaseModel {

  public static table = 'addresses';

  @column({ isPrimary: true })
  public id: number

  @column({ serializeAs: "order_id" })
    public orderId: number

    @column({ serializeAs: "longitude" })
    public longitude: number

    @column({ serializeAs: "latitude" })
    public latitude: number

    @column({ serializeAs: "city" })
    public city: String

    @column({ serializeAs: "country" })
    public country: String

    @column({ serializeAs: "area" })
    public area: String

    @column({ serializeAs: "street" })
    public street: String

    @column({ serializeAs: "building_no" })
    public buildingNo: String
    
  @column({ serializeAs: 'addresses' })
  public addresses: string

  @column({ serializeAs: 'city_id' })
  public cityId: number
  
  @column.dateTime({ autoCreate: true })
  public createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  public updatedAt: DateTime

  @belongsTo(() => City,{
    foreignKey: 'cityId',
  })
  public City:BelongsTo<typeof City>
}


